hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ReconTaskControllerImpl.java

```diff
@@ -416,19 +459,358 @@ private void processEventBatchDirectly(OMUpdateEventBatch events) {
   public boolean hasEventBufferOverflowed() {
     return eventBuffer.getDroppedBatches() > 0;
   }
-  
-  @Override
+
+  /**
+   * Reset the event buffer overflow flag after full snapshot is completed.
+   */
   public void resetEventBufferOverflowFlag() {
     eventBuffer.resetDroppedBatches();
   }
   
   @Override
-  public boolean hasDeltaTasksFailed() {
-    return deltaTasksFailed.get();
+  public boolean hasTasksFailed() {
+    return tasksFailed.get();
+  }
+
+  /**
+   * Reset the task(s) failure flag after reinitialization is completed.
+   */
+  public void resetTasksFailureFlag() {
+    tasksFailed.compareAndSet(true, false);
+  }
+
+  @Override
+  public synchronized ReconTaskController.ReInitializationResult queueReInitializationEvent(
+      ReconTaskReInitializationEvent.ReInitializationReason reason) {
+    LOG.info("Queueing task reinitialization event due to: {} (retry attempt count: {})", reason,
+        eventProcessRetryCount.get());
+
+    ReInitializationResult reInitializationResult = validateRetryCountAndDelay();
+    if (reInitializationResult == ReInitializationResult.MAX_RETRIES_EXCEEDED) {
```

Comment:
it can return retry_later also, if null!= result, then we can return reInitializtionResult